tf2_py/CMakeLists.txt

```diff
@@ -1,26 +1,47 @@
 cmake_minimum_required(VERSION 3.5)
 project(tf2_py)
 
-if(NOT WIN32)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wextra -Wpedantic")
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra)
```

Comment:
We use these flags everywhere in the ros2 codebase, we usually try to also use Wpedantic when it doesn't introduce spurious changes or make us diverge too much for upstream. I don't see a good reason for this package to not do so  What is the rationale for enforcing these flags only in debug mode? Is deployment software allowed to have warnings that debug mide doesn't allow ?